python_modules/dagster/dagster/_core/definitions/decorators/op_decorator.py

```diff
@@ -75,6 +114,44 @@ def __init__(
         self.ins = check.opt_nullable_mapping_param(ins, "ins", key_type=str, value_type=In)
         self.out = out
 
+    def get_code_origin_tags(self, fn: Callable) -> Dict[str, Any]:
+        """Generates the code origin tag for an op. This is a dictionary with a single key, __code_origin,
+        whose value is a string of the form "<file>:<line>" where <file> is the path to the file where
+        the op is defined and <line> is the line number on which the op is defined. This tag is used
+        to link to the location of the op in the user's editor from Dagit.
+        """
+        if not is_code_origin_enabled():
+            return {}
+
+        # Attempt to fetch information about where the op is defined in code,
+        # which we'll attach as a tag to the op
+        cwd = os.getcwd()
+        origin_file: Optional[str] = None
+        origin_line = None
+        try:
```

Comment:
To make it distinguish the Dagster-specific behavior from the underlying origin-scraping implementation, it would be helpful to factor out the latter into a utility function, perhaps in inference.py or some utils file.